tsconfig.json

```diff
@@ -7,9 +7,9 @@
     "jsxFactory": "React.createElement",
     "moduleResolution": "node",
     "noImplicitAny": true,
-    "outDir": "dist",
     "declaration": true,
-    "strict": true
+    "strict": true,
+    "noEmit": true
```

Comment:
Seems kinda confusing to have this option here.
What if we define npm script like that:
```
"check-types": tsc --noEmit true"
```

It's also better because contributors will be able to find this command in package.json (no need to guess that `yarn tsc` exists)

Reply:
`tsc` is always going to be there with Typescript as a dependency though.

Separate script makes sense, but I do think we should keep `noEmit` as without it we'd essentially have two build tools at once. As `tsc` is commonly used in libs as a build tool, o think disabling that functionality of it might reduce chance of mistakes/confusion. 

I can see it either way though.

Reply:
Hm I don't think so? A bundler that would acknowledge TS's noEmit would be rather pointless... if noEmit is false, bundler acts normally. If noEmit true, do absolutely nothing? 

I can't see any reason why a bundler would acknowledge an option like that.

Fair enough though, I'll alter it.